clients/src/main/java/org/apache/kafka/common/requests/UpdateFeaturesResponse.java

```diff
@@ -82,17 +84,31 @@ public static UpdateFeaturesResponse parse(ByteBuffer buffer, short version) {
         return new UpdateFeaturesResponse(new UpdateFeaturesResponseData(new ByteBufferAccessor(buffer), version));
     }
 
-    public static UpdateFeaturesResponse createWithErrors(ApiError topLevelError, Map<String, ApiError> updateErrors, int throttleTimeMs) {
+    public static UpdateFeaturesResponse createWithErrors(short version, ApiError topLevelError, Map<String, ApiError> updateErrors, int throttleTimeMs) {
         final UpdatableFeatureResultCollection results = new UpdatableFeatureResultCollection();
-        for (final Map.Entry<String, ApiError> updateError : updateErrors.entrySet()) {
-            final String feature = updateError.getKey();
-            final ApiError error = updateError.getValue();
-            final UpdatableFeatureResult result = new UpdatableFeatureResult();
-            result.setFeature(feature)
-                .setErrorCode(error.error().code())
-                .setErrorMessage(error.message());
-            results.add(result);
+        Optional<Map.Entry<String, ApiError>> errorEntry = Optional.empty();
+        if (version > 1) {
+            Stream<Map.Entry<String, ApiError>> errorEntries = updateErrors.entrySet().stream().filter(entry ->
```

Comment:
This is interesting, because this code is really only currently called in the ZK path. In the case where it would be called with a partial update, we would potentially want to catch the error? But I can remove if it is confusing.

Reply:
>  In the case where it would be called with a partial update, we would potentially want to catch the error?

Reply:
Yes. I wrote this in case someone would use this in the the future for some Kraft case. But maybe that is not needed.

Reply:
An alternative is to just remove the ability to pass in a collection of updates.

Reply:
Ok -- actually I think the best way forward is to leave as is and not update the createWithErrors method. When we remove the ZK code, we can remove the ability to pass in updateErrors. I can leave a todo for this if it helps.